sources/constraint-definition.asciidoc

```diff
@@ -490,64 +490,37 @@ It is often useful to declare the same constraint more than once to the same tar
 
 Note to constraint designers: each constraint annotation should be coupled with its corresponding multi-valued annotation. The specification recommends, though does not mandate, the definition of an inner annotation named [classname]`List`.
+Each constraint annotation type should be meta-annotated with [classname]`java.lang.annotation.Repeatable`, referencing the corresponding [classname]`List` annotation.
+This marks the constraint annotation type as repeatable and lets users specify the constraint several times without explicitly using the [classname]`List` annotation.
+All built-in annotations follow this pattern.
 
 .Multi-valued constraint definition
 ====
-
-[source, JAVA]
+[source, JAVA, indent=0]
 ----
-/**
- * Validate a zipcode for a given country
- * The only supported type is String
- */
-@Documented
-@Constraint(validatedBy = ZipCodeValidator.class)
-@Target({ METHOD, FIELD, ANNOTATION_TYPE, CONSTRUCTOR, PARAMETER })
-@Retention(RUNTIME)
-public @interface ZipCode {
-
-    String countryCode();
-
-    String message() default "{com.acme.constraint.ZipCode.message}";
-
-    Class<?>[] groups() default {};
-
-    Class<? extends Payload>[] payload() default {};
-
-    /**
-     * Defines several @ZipCode annotations on the same element
-     * @see (@link ZipCode}
-     */
-    @Target({ METHOD, FIELD, ANNOTATION_TYPE, CONSTRUCTOR, PARAMETER })
-    @Retention(RUNTIME)
-    @Documented
-    @interface List {
-        ZipCode[] value();
-    }
-}
+include::{spec-examples-source-dir}org/beanvalidation/specexamples/chapter03/multivaluedconstraint/ZipCode.java[tags=include]
 ----
 
 ====
 
 .Multi-valued constraint declaration
 ====
-
-[source, JAVA]
+[source, JAVA, indent=0]
 ----
-public class Address {
-    @ZipCode.List( {
-            @ZipCode(countryCode="fr", groups=Default.class
-                     message = "zip code is not valid"),
-            @ZipCode(countryCode="fr", groups=SuperUser.class
-                     message = "zip code invalid. Requires overriding before saving.")
-            } )
-    private String zipcode;
-}
+include::{spec-examples-source-dir}org/beanvalidation/specexamples/chapter03/multivaluedconstraint/Address.java[tags=include]
 ----
-
 ====
 
 In this example, both constraints apply to the [methodname]`zipcode` field but with different groups and with different error messages.
+It is also possible to specify a constraint several types by explicitly using the [classname]`@List` annotation (though simply repeating the annotation is the preferred idiom as of Bean Validation 2.0 and Java 8):
+
+.Multi-valued constraint declaration using explicit [classname]`@List` annotation
+====
+[source, JAVA, indent=0]
+----
+include::{spec-examples-source-dir}org/beanvalidation/specexamples/chapter03/multivaluedconstraint/explicitlist/Address.java[tags=include]
```

Comment:
Is it such a good idea to use the chapter number? What if we add a new chapter?

I think it would probably be better to use the name of the .adoc file as the first directory.